### DISCUSSION TITLE
Новые методы и свойства для класса Inventory (SetWeapon(), WeaponId)

### DISCUSSION THREAD

**Discussion by justqstn** [Sokol-Games-GmbH/PixelCombats.PublicApi#156] [Ideas]
Для полей Inventory.Secondary, Inventory.Main, Inventory.Melee, Inventory.Explosive добавить методы SetWeapon() и GetWeapon() (или свойство WeaponId)
Данные методы предназначены только для игрока!

SetWeapon(WeaponId Id) - выдать игроку оружие с айди _Id_. Если у игрока нету данного оружия, то метод вернет false, если есть - выдаст оружие и вернет true
WeaponId - приватное поле класса, хранит в себе айди оружия игрока

Пример 1. Выдаем игроку AWP, если у игрока нет AWP, то выдаем MP-5
// 1 - MP-5, 12 - AWP
if (!player.Inventory.Main.SetWeapon(12)) {
  player.Ui.Hint.Value = "Выдано AWP";
} else {
  player.Inventory.Main.SetWeapon(1);
  player.Ui.Hint.Value = "У вас нет AWP";
}

Пример 2. Проверяем какое оружие у игрока, если это MAC-11, то выдаем ему беретту
// 2 - Beretta, 18 - MAC-11
Teams.OnRequestJoinTeam.Add(function(player, team) {
  if (player.Inventory.Secondary.WeaponId == 18) player.Inventory.Secondary.SetWeapon(2);
});

Будет очень полезно для некоторых режимов, где не хочется видеть например, рпг, или выдавать его только за определенную сумму. Можно будет создать расширенный магазин оружий (как в моем режиме закладки бомбы)

**Comment by ZhenyaMartynov** [comment]
Лучше сделать отдельно метод Contains(WeaponId) для того чтобы просто проверять а не выдавать оружие

**Comment by justqstn** [comment]
> Лучше сделать отдельно метод Contains(WeaponId) для того чтобы просто проверять а не выдавать оружие

лучше не добавлять новый функционал?)

**Reply by ZhenyaMartynov** [reply]
? Почему? Всмысле сделать новый ещё 1 метод к предложенному

**Reply by justqstn** [reply]
зачем этот метод, если будет свойство?

**Reply by ZhenyaMartynov** [reply]
Не понял точно о каком ты свойстве, я про то что можно будет проверить есть ли у игрока оружие, если ты про это конечно

**Reply by justqstn** [reply]
перечитай еще раз то, что я в дискуссии написал)

**Reply by ZhenyaMartynov** [reply]
А понял про что ты
Всё же методом удобнее?
Да и выглядит логичнее
Конечно можно и так

**Comment by forwork10** [comment]
@ZhenyaMartynov тупой...

**Comment by kkohno** [comment]
идея хорошая, сделаем
